internal/mode/static/state/dataplane/configuration.go

```diff
@@ -43,28 +43,163 @@ func BuildConfiguration(
 	}
 
 	baseHTTPConfig := buildBaseHTTPConfig(g)
-	upstreams := buildUpstreams(ctx, g.Gateway.Listeners, resolver, baseHTTPConfig.IPFamily)
+	upstreams := buildUpstreams(ctx, g.Gateway.Listeners, serviceResolver, baseHTTPConfig.IPFamily)
 	httpServers, sslServers := buildServers(g, generator)
+	passthroughServers := buildPassthroughServers(g)
+	streamUpstreams := buildStreamUpstreams(ctx, g.Gateway.Listeners, serviceResolver, baseHTTPConfig.IPFamily)
 	backendGroups := buildBackendGroups(append(httpServers, sslServers...))
 	keyPairs := buildSSLKeyPairs(g.ReferencedSecrets, g.Gateway.Listeners)
 	certBundles := buildCertBundles(g.ReferencedCaCertConfigMaps, backendGroups)
 	telemetry := buildTelemetry(g)
 
 	config := Configuration{
-		HTTPServers:    httpServers,
-		SSLServers:     sslServers,
-		Upstreams:      upstreams,
-		BackendGroups:  backendGroups,
-		SSLKeyPairs:    keyPairs,
-		Version:        configVersion,
-		CertBundles:    certBundles,
-		Telemetry:      telemetry,
-		BaseHTTPConfig: baseHTTPConfig,
+		HTTPServers:           httpServers,
+		SSLServers:            sslServers,
+		TLSPassthroughServers: passthroughServers,
+		Upstreams:             upstreams,
+		StreamUpstreams:       streamUpstreams,
+		BackendGroups:         backendGroups,
+		SSLKeyPairs:           keyPairs,
+		Version:               configVersion,
+		CertBundles:           certBundles,
+		Telemetry:             telemetry,
+		BaseHTTPConfig:        baseHTTPConfig,
 	}
 
 	return config
 }
 
+// buildPassthroughServers builds TLSPassthroughServers from TLSRoutes attaches to listeners.
+func buildPassthroughServers(g *graph.Graph) []Layer4VirtualServer {
+	passthroughServersMap := make(map[graph.L4RouteKey][]Layer4VirtualServer)
+	listenerPassthroughServers := make([]Layer4VirtualServer, 0)
+
+	passthroughServerCount := 0
+
+	for _, l := range g.Gateway.Listeners {
+		if !l.Valid || l.Source.Protocol != v1.TLSProtocolType {
+			continue
+		}
+		foundRouteMatchingListenerHostname := false
+		for key, r := range l.L4Routes {
+			if !r.Valid {
+				continue
+			}
+
+			var hostnames []string
+
+			for _, p := range r.ParentRefs {
+				if val, exist := p.Attachment.AcceptedHostnames[l.Name]; exist {
+					hostnames = val
+					break
+				}
+			}
+
+			if _, ok := passthroughServersMap[key]; !ok {
+				passthroughServersMap[key] = make([]Layer4VirtualServer, 0)
```

Comment:
Not sure I understand why we're using a map here instead of just a slice of servers.

Reply:
I used map because I don't know the number of routes so appending to a slice is slow and adding a new key to a map is faster. But its probably not super significant and if you think it makes the code harder to follow then I can switch to using a slice.

Reply:
I guess that makes sense and is probably fine.